### ISSUE TITLE
[FeatureRequest] Show the type of an exception in the stacktrace, when one occurs

### ISSUE THREAD

**Issue by PhilippMDoerner** [planety/prologue#147]
Heyho everybody,

I am building a normal REST-API with Prologue, norm and jsony, the database being sqlite.

Recently found myself in a scenario where I wanted to configure some HTTP404 responses.
So I first wanted to see what currently happens if I throw a request for a resource at the server that I know doesn't exist. Easily said and done, it causes an error from the database, which leads to the default HTTP500 response in Debug mode being shown.

```
ERROR Record not found
Async traceback:
  /home/isofruit/dev/nimstoryfont/src/nimstoryfont.nim(18)                                        nimstoryfont
  /home/isofruit/dev/nimstoryfont/src/nimstoryfont.nim(15)                                        main
  /home/isofruit/.nimble/pkgs/prologue-0.5.0/prologue/core/application.nim(533)                   run
  /home/isofruit/.nimble/pkgs/prologue-0.5.0/prologue/core/beast/server.nim(31)                   serve

  [... just a very long stacktrace ...]

    /home/isofruit/.choosenim/toolchains/nim-1.6.2/lib/pure/asyncmacro.nim(31)                      switchNimAsyncContinue
    /home/isofruit/.choosenim/toolchains/nim-1.6.2/lib/pure/asyncmacro.nim(131)                     switchIter
    /home/isofruit/.choosenim/toolchains/nim-1.6.2/lib/pure/asyncfutures.nim(389)                   read
  ]#
Exception message: Record not found

DEBUG 500 Internal Server Error {"content-type": @["text/plain; charset=UTF-8"]}
```

What I wanted to do from there then, is wrap a try-catch around the database request in my controller and if I catch an exception of the type that is thrown when there is no record, then I want to return an HTTP404. 

Since the stack trace that is shown when an error occurs doesn't include the *type* of the Exception that was thrown though, I had to write the code that catches the exception and echo's out its type myself and then trigger the scenario that caused the exception.

```nim
  try:
        <some database calling code>
            
    except CatchableError as e:
        echo "Error of type '" & $e.name & "'"
        raise e
```

To reduce that hassle in the future, it would be a really nice convenience feature if we could have the type of an exception show up in the stacktrace that is printed to the console. That way it's easier to see what to catch in a `try-except` block.

**Comment by ringabout** [comment]
Hello @PhilippMDoerner, nice to meet you!

It seems to be a nice addition, feel free to make a change here: https://github.com/planety/prologue/blob/4dcb7f4c4bf37cdafef15884560317530b69a7ff/src/prologue/core/application.nim#L469 (for instance logging the name of exception together with exception messages)

I contribute to Nim on weekdays, so there are only weekends left for me to contribute to Prologue. If you have any problem, feel free to ask me on this issue or Nim discord (same name). I can make a PR next week.

**Comment by PhilippMDoerner** [comment]
@xflywind We actually already chatted in discord, I go by Isofruit there :smile: 

Given that it should be a fairly simple PR I can throw one together as well.

I'd swap the line you pointed out merely to this one:

```nim
 logging.error $e.name & ": " & e.msg 
```

What I think might be even more useful though, would be also printing the package that an exception comes from (in my case e.g. the `NotFoundError` came from norm, which I hadn't realized). That way you'd know where to import that exception-type from. How to do that (and if there's a simple way to do that) I'm still trying to figure out though.

**Comment by ringabout** [comment]
> We actually already chatted in discord, I go by Isofruit there 😄

Haha, I see.

> I'd swap the line you pointed out merely to this one:

Looks good enough.

> What I think might be even more useful though, would be also printing the package that an exception comes from (in my case e.g. the NotFoundError came from norm, which I hadn't realized). That way you'd know where to import that exception-type from. How to do that (and if there's a simple way to do that) I'm still trying to figure out though.

It sounds a good idea. I replied you at discord (I don't know how to achieve that at runtime either).

**Comment by PhilippMDoerner** [comment]
> It sounds a good idea. I replied you at discord (I don't know how to achieve that at runtime either).

I like this idea very much as well, sadly though I also couldn't find a way. I went through the standard libraries as well, but nothing jumps out. You already posted a very valid way of receiving the "owner" of a type at compile time through a macro (for documentation purposes I have attached that down below if this issue ever gets picked up again), but for getting that at runtime that's a different story. And other than recording ownership at compile time for all types and then accessing that at runtime, which is way too much complexity than this "nice-to-have" feature deserves, I have no ideas for that either.

I'll make a PR with the suggestion I made.

```nim
import macros

macro getOwnerTypeStringImpl(a: typed): string =
  let n = a.owner
  var ret: string
  case n.symKind
  of nskModule:
    ret = n.repr
  else:
    doAssert false
  newLit ret

echo getOwnerTypeStringImpl(ValueError)
```
